File: src/main/java/leetcode/tree/DeepestLeavesSum.java
package leetcode.tree;

import java.util.LinkedList;
import java.util.Queue;

public class DeepestLeavesSum {
    public int deepestLeavesSum(TreeNode root) {

        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int ret = 0;
        while (!q.isEmpty()) {
            int nodeSize = q.size();
            ret = 0;

            for (int i = 0; i < nodeSize; i++) {
                TreeNode n = q.poll();
                ret = ret + n.val;

                if (n.left != null) {
                    q.add(n.left);
                }
                if (n.right != null) {
                    q.add(n.right);
                }
            }

        }

        return ret;
    }
}

File: src/main/python/leetcode/tree/deepest_leaves_sum.py


class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:

        q = [root]
        ret = 0
        while len(q) != 0:
            size = len(q)
            ret = 0
            for _ in range(0, size):
                n = q.pop(0)
                ret = ret + n.val

                if n.left is not None:
                    q.append(n.left)
                if n.right is not None:
                    q.append(n.right)

        return ret
